lib/bot.js

```diff
@@ -183,7 +193,19 @@ class Bot {
       }
 
       const withMentions = text.replace(/@[^\s]+\b/g, (match) => {
-        const user = this.discord.users.find('username', match.substring(1));
+        const search = match.substring(1);
+        const guild = discordChannel.guild;
+        let user = guild.members.find('nickname', search);
+        if (!user) {
+          user = this.discord.users.find('username', search);
+          if (user) {
+            const serverUser = guild.members.find('id', user.id);
+            if (serverUser.nickname && serverUser.nickname !== search) {
+              // User has a guild nickname which is different from the search query
```

Comment:
@ekmartin I'm not sure that's a good idea. The people on IRC would be unable to see what the usernames of users on Discord are, and could possibly even inadvertently mention someone they weren't trying to mention.